### PR TITLE
fixes #176 by introducing a soft reset routine

### LINKED ISSUE #176: A soft reset routine for the CCLO should be introduced and applied during initialization
Currently, the ACCL initialization routine on the host verifies that CCLO_ADDR::CFGRDY_OFFSET equals 0, before performing any other steps, and throws an error otherwise. In case of software crashes occurring after the initialization stage and before the execution of the operation, e.g., for a copy, the value of CCLO_ADDR::CFGRDY_OFFSET is different from 0, and any other attempts to re-initialize the CCLO fail. 
It is therefore necessary to introduce a soft reset routine, invoked during initialization, to set it back to 0 and allow any other configuration steps to occur, without throwing any exceptions.

### PR DESCRIPTION
**Pull request by mar-ven**

### CODE REVIEW

File: driver/xrt/include/accl.hpp [hunk 1]
```diff
@@ -113,6 +113,11 @@ class ACCL {
    */
   ~ACCL();
 
+  /**
+   * Performs a soft reset of the CCLO.
+  */
+  void reset();
```
Comment (quetric) [inline driver/xrt/include/accl.hpp:119]: call it soft_reset please

File: driver/xrt/src/accl.cpp [hunk 2]
```diff
@@ -80,10 +80,8 @@ ACCL::~ACCL() {
   delete cclo;
 }
 
-void ACCL::deinit() {
-  debug("Removing CCLO object at " + debug_hex(cclo->get_base_addr()));
-
-  cclo->printDebug();
+void ACCL::reset() {
```
Comment (quetric) [inline driver/xrt/src/accl.cpp:83]: out of alignment with the surrounding code

**Review by quetric — CHANGES_REQUESTED** [review]

File: driver/xrt/src/accl.cpp (hunk 2 above)
Comment (mar-ven) [reply driver/xrt/src/accl.cpp]: The indentation is consistent

File: driver/xrt/include/accl.hpp (hunk 1 above)
Comment (mar-ven) [reply driver/xrt/include/accl.hpp]: Modifications performed

**Review by quetric — APPROVED** [review]

**Pull request merged**